### ISSUE TITLE
RuntimeError: vLLM cannot currently support max_model_len=65536 with block_size=16 on GPU with compute capability (8, 9) (required shared memory 264252.0 > available shared memory 101376).

### ISSUE THREAD

**Issue by asifehmad** [vllm-project/vllm#1267]
I tried running this [model](https://huggingface.co/Open-Orca/LlongOrca-7B-16k) on **1 x RTX 6000 Ada** as:
```
from vllm import LLM, SamplingParams
sampling_params = SamplingParams(temperature=0.8, top_p=0.95)
llm = LLM(model="Open-Orca/LlongOrca-7B-16k")
```
But getting this below error once the model loading completes.

```
INFO 10-05 12:28:40 llm_engine.py:205] # GPU blocks: 3028, # CPU blocks: 512
Traceback (most recent call last): 
File "<stdin>", line 1, in <module>  
File "/env/lib/python3.10/site-packages/vllm/entrypoints/llm.py", line 89, in __init__    
     self.llm_engine = LLMEngine.from_engine_args(engine_args)  
File "/env/lib/python3.10/site-packages/vllm/engine/llm_engine.py", line 229, in from_engine_args    
     engine = cls(*engine_configs,  
File "/env/lib/python3.10/site-packages/vllm/engine/llm_engine.py", line 111, in __init__    self._init_cache()  
File "/env/lib/python3.10/site-packages/vllm/engine/llm_engine.py", line 217, in _init_cache
    self._run_workers("init_cache_engine", cache_config=self.cache_config)
File "/env/lib/python3.10/site-packages/vllm/engine/llm_engine.py", line 692, in _run_workers
    output = executor(*args, **kwargs)
File "/env/lib/python3.10/site-packages/vllm/worker/worker.py", line 147, in init_cache_engine
    _check_if_can_support_max_seq_len(max_seq_len, self.block_size)
  File "/env/lib/python3.10/site-packages/vllm/worker/worker.py", line 381, in _check_if_can_support_max_seq_len
    raise RuntimeError(
RuntimeError: vLLM cannot currently support max_model_len=65536 with block_size=16 on GPU 
with compute capability (8, 9) (required shared memory 264252.0 > available shared memory 101376). 
This will be fixed in a future release.

```

Can anyone please check it?

Thanks!

**Comment by WoosukKwon** [comment]
Hi @asifehmad, currently due to a limitation in our attention implementation, there is a limit in the maximum sequence length supported by a GPU. It seems your GPU does not have enough shared memory to support the 16K sequence length. We will fix this and eliminate the limit.

If you wish to use the model right now, you have to manually limit the maximum length by `LLM(..., max_model_len=8192)`.

**Comment by asifehmad** [comment]
> Hi @asifehmad, currently due to a limitation in our attention implementation, there is a limit in the maximum sequence length supported by a GPU. It seems your GPU does not have enough shared memory to support the 16K sequence length. We will fix this and eliminate the limit.
> 
> If you wish to use the model right now, you have to manually limit the maximum length by `LLM(..., max_model_len=8192)`.

Thank you!

**Comment by bonuschild** [comment]
> Hi @asifehmad, currently due to a limitation in our attention implementation, there is a limit in the maximum sequence length supported by a GPU. It seems your GPU does not have enough shared memory to support the 16K sequence length. We will fix this and eliminate the limit.
> 
> If you wish to use the model right now, you have to manually limit the maximum length by `LLM(..., max_model_len=8192)`.

How is that? Is it fixed yet? I encounter it again :)

**Comment by WoosukKwon** [comment]
@bonuschild This was fixed by #1510, and will be included in the next release.
